go/vt/vtgate/router.go

```diff
@@ -75,7 +75,8 @@ func (rtr *Router) MessageAck(ctx context.Context, keyspace, name string, ids []
 	if err != nil {
 		return 0, err
 	}
-	vcursor := newQueryExecutor(ctx, topodatapb.TabletType_MASTER, nil, nil, rtr)
+	// TODO(sougou): Change this to use Session.
```

Comment:
This is intended. We have to eventually upgrade all methods to compute the target from the Session.